### ISSUE TITLE
Meaning of scores in the log_eval_score file and how to export the predicted labels 

### ISSUE THREAD

**Issue by zahrakhanjani128** [asvspoof-challenge/2021#22]
I applied the pre-trained DF baseline model (B03) on the toy dataset and have an output named log_eval_score. What are the meaning of the scores? How can I see the predicted label for each audio clip?

**Comment by TonyWangX** [comment]
Hi, according to https://github.com/asvspoof-challenge/2021/blob/aae41bb3f4f38795ebf6901c7bbf3461e4ad662b/DF/Baseline-LFCC-LCNN/project/README#L94

The three columns means trial name, label, and CM score.

```2. Output score has this following format:

   Output, File name, label, score
   Output, LA_E_8688127, 0, -0.011127

   If label of the trial is given in the protocol file, the label will be: 
     0, which denotes spoof
     1, which denotes bona fide

   If label of the trial is not given in the protocol file, the label will be:
    -1, which means unknown (not provided)
```

**Comment by zahrakhanjani128** [comment]
Hi, thank you for your response. 
I used "02_eval_alternative.sh", so there is no need for protocol.txt file (am I right?). The output that I got is only the name of the audio file and score. 
Even if I provide the protocol file, it will be the true labels (am I right?). Then, I need the predicted labels, how should I export those?

**Comment by TonyWangX** [comment]
> Then, I need the predicted labels, how should I export those?
Do you mean the binary labels? 

Just to clarify:

If the protocol is provided, the printed log-**.txt will show the **ground truth** labels {0, 1}. They are not predicted labels

If the protocol is not provided, the printed log-**.txt will show a dummy placeholder -1. 

In the current script, the verbose log-.txt will be simplified into  log-_score.txt, which contains two columns: trial and score.  (sorry that log_score.txt has two columns, not three)

Yes, there is no need to give protocol.txt. The code will skip loading it.

The score is a real-valued number that indicates how likely the trial is bona fide. The higher the value is, the more likely that the trial is bona fide.  (See more in the notebook below)

**Comment by TonyWangX** [comment]
To get the predicted labels,  you need to write a simple script by yourself. 
```
specify a classification threshold \theta
if score > \theta, the trial is bonafide
else, the trial is spoofed
```

We don't output the binary labels since they are not required by the challenge evaluation plan.
The challenge evaluation is done by sweeping the threshold and computing an equal error rate. 
(see https://github.com/nii-yamagishilab/project-NN-Pytorch-scripts/blob/master/tutorials/b2_anti_spoofing/chapter_a0_EER.ipynb)

**Comment by zahrakhanjani128** [comment]
Many thanks for the clarification! It is really helpful. 
I am not using the metrics of the challenge in my current work. Therefore, it is correct to use a variety of different thresholds in my case, and see which one works best for my dataset. Right?

**Comment by TonyWangX** [comment]
That could be one way to go, but how to set the threshold depends on the application. 

I personally don't know more about the topic, sorry : <

**Comment by zahrakhanjani128** [comment]
Thank you very much! The guidance and information are very appreciated! I am closing this issue.
